pliers/extractors/audio.py

```diff
@@ -488,4 +494,140 @@ class PercussiveExtractor(LibrosaFeatureExtractor):
+    yamnet_kwargs (optional): Optional named arguments that modify input 
+        parameters for the model (see params.py file in yamnet repository)
+    '''
+
+    _log_attributes = ('hop_size', 'top_n', 'label_subset', 'weights_path',
+                       'yamnet_kwargs')
+
+    def __init__(self, hop_size=0.1, top_n=None, label_subset=None,
+                 weights_path=None, **yamnet_kwargs):
+        try:
+            verify_dependencies(['yamnet'])
+        except MissingDependencyError: 
+            raise MissingDependencyError(dependencies=None,
+                                         custom_message=YAMNET_INSTALL_MESSAGE)
+        verify_dependencies(['tensorflow'])
+
+        MODULE_PATH = path.dirname(yamnet.__file__)
+        LABELS_PATH = path.join(MODULE_PATH, 'yamnet_class_map.csv')
+        self.weights_path = weights_path or path.join(MODULE_PATH, 'yamnet.h5')
+        self.hop_size = hop_size
+        self.yamnet_kwargs = yamnet_kwargs
+        self.params = yamnet.params
+        self.params.PATCH_HOP_SECONDS = hop_size
+        for par, v in self.yamnet_kwargs.items():
+            if par in self.params.__dict__:
+                setattr(self.params, par, v)
+        self.labels = pd.read_csv(LABELS_PATH)['display_name'].tolist()
+        self.label_subset = label_subset
+        self.top_n = top_n
+        if self.label_subset:
+            for l in self.label_subset:
+                if l not in self.labels:
+                    logging.warning('''Label {} does not exist. 
+                                     Dropping.'''.format(l))
+        super(AudiosetLabelExtractor, self).__init__()
+
+    def _extract(self, stim):
+        params = self.params
+        params.SAMPLE_RATE = stim.sampling_rate
+
+        if params.SAMPLE_RATE >= 2 * params.MEL_MAX_HZ:
+            if params.SAMPLE_RATE != 16000:
+                logging.warning(
+                    'The sampling rate of the stimulus is '
+                    f'{params.SAMPLE_RATE}Hz. '
+                    'YAMNet was trained on audio sampled at 16000Hz. ' 
+                    'This should not impact predictions, but you can resample ' 
+                    'the input using AudioResamplingFilter for full conformity ' 
+                    'to training.')
+            if params.MEL_MIN_HZ != 125 or params.MEL_MAX_HZ != 7500:
+                logging.warning(
+                    'Custom values for MEL_MIN_HZ and MEL_MAX_HZ '
+                    'were passed. Changing these defaults might affect ' 
+                    'model performance.')
+        else:
+            raise ValueError(
+                f'The sampling rate of your stimulus ({params.SAMPLE_RATE}Hz)'
+                ' must be at least twice the value of MEL_MAX_HZ '
+                f'({params.MEL_MAX_HZ}Hz). ' 
+                'Upsample your audio stimulus (recommended) or pass a lower '
+                'value of MEL_MAX_HZ when initializing this extractor.')
+
+        labels = self.labels
```

Comment:
Is this necessary? it refers to the same variable anyways, so maybe just refer to `self.labels` further down.